pkg/handler/collector/git/git.go

```diff
@@ -0,0 +1,179 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package git_collector
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/guacsec/guac/pkg/handler/collector"
+	"github.com/guacsec/guac/pkg/handler/collector/file"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/logging"
+	"go.uber.org/zap"
+)
+
+const (
+	CollectorGit = "GIT"
+)
+
+type gitCol struct {
+	url           string
+	dir           string
+	lastChecked   time.Time
+	poll          bool
+	interval      time.Duration
+	fileCollector *file.FileCollector
+}
+
+func NewGitCol(ctx context.Context, url string, dir string, poll bool, interval time.Duration) *gitCol {
+	logger := logging.FromContext(ctx)
+	fileCollector := file.NewFileCollector(ctx, dir, false, time.Second)
+	err := collector.RegisterDocumentCollector(fileCollector, file.FileCollectorType)
+	if err != nil {
+		logger.Errorf("unable to register file collector: %v", err)
```

Comment:
this should return an error so that we know that the file collector has not been registered.

Reply:
We removed the registration, going to push the changes soon.

Reply:
hmm, did you intend to have these changes pushed yet?